names/names.py

```diff
@@ -10,13 +10,44 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
+# Provided Solution - Runtime O(n^n)
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
+for name_1 in names_1: # O(n)
+    for name_2 in names_2: # O(n)
         if name_1 == name_2:
             duplicates.append(name_1)
 
 end_time = time.time()
+print ("\n\nProvided Solution")
 print (f"{len(duplicates)} duplicates:\n\n{', '.join(duplicates)}\n\n")
 print (f"runtime: {end_time - start_time} seconds")
 
+## ________ My solution section ____________
+start_time = time.time()
+
+f = open('names_1.txt', 'r')
+names_1 = f.read().split("\n")  # List containing 10000 names
+f.close()
+
+f = open('names_2.txt', 'r')
+names_2 = f.read().split("\n")  # List containing 10000 names
+f.close()
+
+# My Solution - Runtime O(5n) 
```

Comment:
Interesting solution, if you already decided to use set, I'd suggest you to find a cleaner solution using them, sets will improve your runtime complexity, but you could use a lot less logic there. Anyways, it's working like a charm so great job here too.